### ISSUE TITLE
Not creating required files on 1.17.1

### ISSUE THREAD

**Issue by MrEMan** [mircokroon/minecraft-world-downloader#220]
**Describe the bug**
I am using Ubuntu. The program doesn't seem to be able to create the required files blocks.json and registries.json. 1.16+ works great.

**Steps to reproduce**
Run java -jar world-downloader.jar -s server.ip
Watch console fill up with errors :(

This gives a hint "Main has been compiled by a more recent version of the Java Runtime (class file version 60.0), this version of the Java Runtime only recognizes class file versions up to 55.0"

**Log output**
```
Looks like we have not run in version 1.17 before.
Downloading this version's server.jar (https://launcher.mojang.com/v1/objects/0a269b5f2c5b93b1712d0f5dc43b6182b9ab254e/server.jar)
We'll generate some reports now, this may take a minute.
Starting output of Minecraft server.jar:
	Error: LinkageError occurred while loading main class net.minecraft.data.Main
		java.lang.UnsupportedClassVersionError: net/minecraft/data/Main has been compiled by a more recent version of the Java Runtime (class file version 60.0), this version of the Java Runtime only recognizes class file versions up to 55.0
Completed generating reports!
java.io.FileNotFoundException: /home/trevor/Downloads/WorldDownloader/cache/1_17/registries.json (No such file or directory)
	at java.base/java.io.FileInputStream.open0(Native Method)
	at java.base/java.io.FileInputStream.open(FileInputStream.java:211)
	at java.base/java.io.FileInputStream.<init>(FileInputStream.java:153)
	at game.data.registries.RegistryLoader.generateEntityNames(RegistryLoader.java:204)
	at config.Config.loadVersionRegistries(Config.java:265)
	at config.Config.lambda$getGameProtocol$0(Config.java:254)
	at java.base/java.lang.Thread.run(Thread.java:831)
java.io.FileNotFoundException: /home/trevor/Downloads/WorldDownloader/cache/1_17/blocks.json (No such file or directory)
	at java.base/java.io.FileInputStream.open0(Native Method)
	at java.base/java.io.FileInputStream.open(FileInputStream.java:211)
	at java.base/java.io.FileInputStream.<init>(FileInputStream.java:153)
	at game.data.registries.RegistryLoader.generateGlobalPalette(RegistryLoader.java:216)
	at game.data.chunk.palette.GlobalPaletteProvider.requestPalette(GlobalPaletteProvider.java:46)
	at game.data.chunk.palette.GlobalPaletteProvider.getGlobalPalette(GlobalPaletteProvider.java:28)
	at game.data.chunk.Chunk.getBlockStateAt(Chunk.java:320)
	at game.data.chunk.Chunk.getBlockStateAt(Chunk.java:311)
	at game.data.chunk.ChunkEntities.addTileEntity(ChunkEntities.java:108)
	at game.data.chunk.ChunkEntities.addTileEntity(ChunkEntities.java:142)
	at game.data.chunk.Chunk.parseTileEntities(Chunk.java:285)
	at game.data.chunk.version.Chunk_1_17.parse(Chunk_1_17.java:70)
	at game.data.chunk.ChunkFactory.parseChunk(ChunkFactory.java:108)
	at game.data.chunk.ChunkFactory.readChunkDataPacket(ChunkFactory.java:125)
	at game.data.chunk.ChunkFactory.parse(ChunkFactory.java:85)
	at game.data.chunk.ChunkFactory.lambda$addChunk$1(ChunkFactory.java:72)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
	at java.base/java.lang.Thread.run(Thread.java:831)
java.lang.NullPointerException: Cannot invoke "game.data.chunk.palette.GlobalPalette.getState(int)" because the return value of "game.data.chunk.palette.GlobalPaletteProvider.getGlobalPalette(int)" is null
	at game.data.chunk.Chunk.getBlockStateAt(Chunk.java:320)
	at game.data.chunk.Chunk.getBlockStateAt(Chunk.java:311)
	at game.data.chunk.ChunkEntities.addTileEntity(ChunkEntities.java:108)
	at game.data.chunk.ChunkEntities.addTileEntity(ChunkEntities.java:142)
	at game.data.chunk.Chunk.parseTileEntities(Chunk.java:285)
	at game.data.chunk.version.Chunk_1_17.parse(Chunk_1_17.java:70)
	at game.data.chunk.ChunkFactory.parseChunk(ChunkFactory.java:108)
	at game.data.chunk.ChunkFactory.readChunkDataPacket(ChunkFactory.java:125)
	at game.data.chunk.ChunkFactory.parse(ChunkFactory.java:85)
	at game.data.chunk.ChunkFactory.lambda$addChunk$1(ChunkFactory.java:72)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
	at java.base/java.lang.Thread.run(Thread.java:831)
Chunk could not be parsed!

...

```

**Version**
 - Game: Minecraft 1.17.1
 - Downloader: latest

**Comment by mircokroon** [comment]
You probably have an old Java version installed, you'll have to upgrade to at least Java version 16. You might have multiple different versions or something as well, I'm not sure.

**Comment by MrEMan** [comment]
I do have 3 versions installed. 11, 16 and the latest 17. I've tried with 16 and 17. Unfortunately I still need 11 for another program.

I've used `/opt/jdk/jdk-17.28/bin/java -jar world-downloader.jar` with -s and the server ip and the java 16 command. This works perfectly for 1.16.5.

Regards

**Comment by MrEMan** [comment]
I copied the cache files from 1_16_2 into 1_17 and the download kinda works. The shape of everything is okay but the blocks are totally wrong.

**Comment by mircokroon** [comment]
The program uses the default `java` to generate reports on Block IDs etc. If it's not possible to fix the Java version so that `java` refers to 16+, it won't be able to do that.

One solution might be to make them manually, you'll have to download the corresponding `server.jar` (might already have been downloaded for you if it failed halfway through), then run  `java -cp server.jar net.minecraft.data.Main --reports`. When that's done you have to move the files to get this file structure:

```
.../world-downloader.jar
.../cache/1_17/blocks.json
.../cache/1_17/registries.json
```

**Comment by MrEMan** [comment]
Thank you. The manual solution works.

I may have to tinker with my java installs

Regards

**Comment by OliverParoczai** [comment]
The new 1.18.0 update breaks older versions, causing the same error since it tries to extract 1.17.1 files from the 1.18.0 server jar. I had to checkout the a3dd3e167b791ed13c513eb0edbd840a88e4d0ec commit for the error to go away, since even when placing the 1.17.1 files I downloaded into the 1_18 folder, the map format was incompatible since the code still thought the game was in 1.18.0. After going back to the commit I mentioned, versions under 1.18 started to work.
